pkg/frontend/transport/handler_test.go

```diff
@@ -207,3 +208,41 @@ func TestHandler_FailedRoundTrip(t *testing.T) {
 		})
 	}
 }
+
+func TestHandler_InsightKey(t *testing.T) {
+	var buf concurrency.SyncBuffer
+	logger := log.NewLogfmtLogger(&buf)
+
+	cfg := HandlerConfig{QueryStatsEnabled: true, MaxBodySize: 1024}
+	form := url.Values{
+		"query": []string{"some_metric"},
+		"time":  []string{"42"},
+	}
+	request := httptest.NewRequest("POST", "/api/v1/query", strings.NewReader(form.Encode()))
+	request.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	roundTripper := roundTripperFunc(func(req *http.Request) (*http.Response, error) {
+		assert.NoError(t, req.ParseForm())
+
+		return &http.Response{
+			StatusCode: http.StatusOK,
+			Body:       io.NopCloser(strings.NewReader("{}")),
+		}, nil
+	})
+
+	reg := prometheus.NewPedanticRegistry()
+	handler := NewHandler(cfg, roundTripper, logger, reg)
+
+	req := request.WithContext(user.InjectOrgID(context.Background(), "12345"))
+	resp := httptest.NewRecorder()
+
+	handler.ServeHTTP(resp, req)
+	_, _ = io.ReadAll(resp.Body)
+	require.Equal(t, resp.Code, http.StatusOK)
+
+	assert.Eventually(t, func() bool {
+		logs := buf.String()
+		return strings.Contains(logs, "insight=true")
+		// Ensure that the dispatcher component emits logs with a "true" insight key,
+		// identifying these logs to be exposed to end users via the usage insights system.
+	}, 3*time.Second, time.Second)
```

Comment:
Nit: this is most likely instant, so no need to wait so long. 
```suggestion
	}, time.Second, time.Millisecond)
```